docs/internals/optimizer.rst

```diff
@@ -26,6 +26,10 @@ One can use ``solc --ir-optimized --optimize`` to produce an
 optimized Yul IR for a Solidity source. Similarly, one can use ``solc --strict-assembly --optimize``
 for a stand-alone Yul mode.
 
+.. note::
+    The `PeepholeOptimizer <https://en.wikipedia.org/wiki/Peephole_optimization>`_ is always
+    enabled by default and can only be turned off via the :ref:`JSON-input-output interface <compiler-api>`.
```

Comment:
Peephole optimizer is not the only such step. The other one is the inliner:
>        // The inliner is always on if no details are given,
>        // use details to switch it off.
>        "inliner": true,

Can you mention it too?

Reply:
Done.